app/views/result_filters/location/_form.html.erb

```diff
@@ -19,11 +19,11 @@
         <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
           <h1 class="govuk-fieldset__heading">Find courses by location or by training provider</h1>
         </legend>
-        <div class="govuk-form-group<%= " govuk-form-group--error" if flash[:error] %>" id="search-options">
+        <div class="govuk-form-group" id="search-options">
           <% if flash[:error] and location_error? === false and provider_error? === false%>
             <p class="govuk-error-message">
               <% if flash[:error].kind_of?(Array) %>
-                <%= flash[:error].first %>
+                <%= flash[:error].last %>
```

Comment:
This fixes a bug where the error message in the summary and next to the field is currently sometimes different. This happens when the error message applies to a conditionally-revealed field, where the code uses an array of two items to identify which field was in error, with the first item in the array being the label of the field, and the second item being the actual error message. So the last item of the array should be shown instead of the first one.

See example of the issue:


<img width="807" alt="Screenshot 2021-01-04 at 15 52 05" src="https://user-images.githubusercontent.com/30665/103553064-cdd7b100-4ea4-11eb-8415-80565dad21e9.png">

An array within the flash isn't the ideal way to store these errors, so we should come back and fix this as part of a wider refactor.